### PR TITLE
fix: add repository metadata to all packages

### LINKED ISSUE #3612: package.json repository/directory metadata incomplete
## 🐛 Bug Report

The `package.json` files within packages/* contain incomplete metadata for source repo and source directory. e.g. as according to this advice from [npm's `package.json` documentation](https://docs.npmjs.com/files/package.json):

image

### Have you read the [Contributing Guidelines on issues]

Yes

## To Reproduce

View any `package.json` within `packages/*`. I think all are missing the `directory` field and some are also missing the source repository information altogether (e.g. [@docusaurus/preset-classic](https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-preset-classic/package.json))

## Expected behavior

Monorepo `package.json` files should all contain a valid `repository` object including `directory` field.

I can try correcting this if you agree to receive a PR.

## Actual Behavior

N/A

## Your Environment

N/A

## Reproducible Demo

N/A

### PR DESCRIPTION
**Pull request by rarkins**
Closes #3612

## Motivation

Ensure all npm packages now and in the future contain accurate repository metadata.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

This PR includes its own test, so I think just code inspection is OK.

## Related PRs

None

### CODE REVIEW

File: packages/__tests__/validate-package-json.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+import util from 'util';
+import globCb from 'glob';
+import fsCb from 'fs';
+
+const glob = util.promisify(globCb);
+const readFile = util.promisify(fsCb.readFile);
+
+describe('packages', () => {
```
Comment (rarkins) [inline packages/__tests__/validate-package-json.spec.ts:15]: Feel free to move/refactor/etc this test. Without it, things are likely to get out of sync again.

**Review by lex111 — APPROVED** [review]
LGTM :+1:

**Comment by lex111** [comment]
>   9:20  error  'glob' should be listed in the project's dependencies. Run 'npm i -S glob' to add it  import/no-extraneous-dependencies

There is one error to fix.

**Comment by rarkins** [comment]
@lex111 it's in devDependencies, which is where I think it belongs. Maybe this lint error is because it's in `packages/*` and could be relocated?

**Comment by rarkins** [comment]
Sorry, I'm not sure how to fix this. Jest spec files shouldn't have a `package.json`, and `glob` shouldn't be added to the root `package.json` either. I tried moving the file into `jest/` and also adding a eslint ignore rule, but nothing I tried stops the error message.

**Comment by slorber** [comment]
Thanks,  seems to be passing now ;)

File: package.json [hunk 2]
```diff
@@ -100,6 +100,7 @@
     "eslint-plugin-jsx-a11y": "^6.2.3",
     "eslint-plugin-react": "^7.20.0",
     "eslint-plugin-react-hooks": "^4.0.4",
+    "glob": "^7.1.6",
```
Comment (rarkins) [inline package.json:103]: Thanks. I was sure I'd add this already. Spent hours thinking it wanted me to move the dependency into `dependencies` and trying to fix that.. ugh

Comment (slorber) [reply package.json:103]: :) maybe you didn't use the -W option to add dev dependencies in monorepo root ;)

**Pull request merged**